### PR TITLE
Webhook Label Selector

### PR DESCRIPTION
**Pull request by ionysos**
### Description

This enables the use of webhook label selectors to be able to reduce the load on the cluster for excluded namespaces / objects.
Webhook Label Selectors are available in Kubernetes v1.15+ (https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-objectselector).

### Checklist
Before submitting this PR, please make sure:
- [x] your code builds clean with `make`
- [x] your code lets succeed unit tests with `make test`
- [x] your code lets succeed integration tests
- [x] you have documented new or changed features

### CODE REVIEW

File: docs/features.md [hunk 1]
```diff
@@ -98,3 +98,10 @@ The feature defaults a service account's `automountServiceAccountToken` to false
 |7| **not defined** | false | false | false |
 |8| true | not defined | true | true |
 |9| false | not defined | false | false |
+
+## Karydia Exclusion Handling
+
+Namespaces and other objects can be opted out of being "watched" by karydia. Therefor, there are two options:
```
Comment (Neumann-Nils) [inline docs/features.md]: Typo on "therfor". Should be "therefore".

File: docs/features.md [hunk 2]
```diff
@@ -98,3 +98,10 @@ The feature defaults a service account's `automountServiceAccountToken` to false
...
+- (nearly) each feature provides its own annotation for namespaces and / or other objects to be ignored by the respective karydia feature - see feature descriptions above
```
Comment (Neumann-Nils) [inline docs/features.md]: Would not add a space between "and / or".

File: docs/features.md [hunk 3]
```diff
@@ -98,3 +98,10 @@ The feature defaults a service account's `automountServiceAccountToken` to false
...
+- [values.yaml](../install/charts/values.yaml), which provides karydia (component) installation configurations, provides two blocks called `exclusionNamespaceLabels` and `exclusionObjectLabels`. These blocks define either namespace or other object labels. If they are matched by either namespace or object the karydia webhooks filter them out and, thus, they get fully excluded / ignored by karydia. These settings need to be adjusted before running the installation of karydia.
+:warning: Karydia's network policy feature works differently, without the use of webhooks and, hence, this feature is independent from that configuration setting.
```
Comment (Neumann-Nils) [inline docs/features.md]: Would put this remark into a new line.

File: docs/features.md [hunk 4]
```diff
@@ -98,3 +98,10 @@ The feature defaults a service account's `automountServiceAccountToken` to false
 |7| **not defined** | false | false | false |
 |8| true | not defined | true | true |
 |9| false | not defined | false | false |
+
+## Karydia Exclusion Handling
+
+Namespaces and other objects can be opted out of being "watched" by karydia. Therefor, there are two options:
+- (nearly) each feature provides its own annotation for namespaces and / or other objects to be ignored by the respective karydia feature - see feature descriptions above
+- [values.yaml](../install/charts/values.yaml), which provides karydia (component) installation configurations, provides two blocks called `exclusionNamespaceLabels` and `exclusionObjectLabels`. These blocks define either namespace or other object labels. If they are matched by either namespace or object the karydia webhooks filter them out and, thus, they get fully excluded / ignored by karydia. These settings need to be adjusted before running the installation of karydia.
```
Comment (Neumann-Nils) [inline docs/features.md]: Would not add a space between "excluded / ignored"

**Comment by ionysos** [comment]
> Feature works fine for me. However, I find it kind of confusing to add the label `karydia` if you want to exclude this namespace or object from karydia. Would another label (e.g. `excludeKarydia`) make the functionality more clear?
> 
> Moreover, double check the `features.md`.

Hmm. The idea was that we only have ONE place where to change the settings. If we name the label `excludeKarydia` we have to add an additional label to the karydia app installation. I think this would generate too much overhead. That's the reason why I just reused the currenty available `app: karydia` label.

File: docs/features.md (hunk 1 above)
Comment (ionysos) [reply docs/features.md]: done

File: docs/features.md (hunk 2 above)
Comment (ionysos) [reply docs/features.md]: done

File: docs/features.md (hunk 3 above)
Comment (ionysos) [reply docs/features.md]: done

File: docs/features.md (hunk 4 above)
Comment (ionysos) [reply docs/features.md]: done

**Pull request merged**